<a href="https://colab.research.google.com/github/Beelzebub0126/CSEL_302_BSCS_2B_CABREZA_REN_G/blob/main/2B_CABREZA_EXER5.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd
import numpy as np
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import accuracy_score, confusion_matrix, classification_report
from sklearn.preprocessing import StandardScaler

In [ ]:
df = pd.read_csv("BankCustomerData.csv")

FileNotFoundError: [Errno 2] No such file or directory: 'BankCustomerData.csv'

In [ ]:
df_dummies = pd.get_dummies(df, drop_first=True)

In [ ]:
df['term_deposit'] = df['term_deposit'].map({'yes': 1, 'no': 0})

In [ ]:
df_dummies['term_deposit'] = (df['balance']>50).astype(int)

In [ ]:
X = df_dummies.drop(['age', 'balance', 'day', 'duration', 'campaign', 'pdays', 'previous', 'term_deposit'],axis=1)
Y = df_dummies['term_deposit']

In [ ]:
X_train, X_test, Y_train, Y_test = train_test_split(X,Y, test_size=0.2,random_state=42)

In [ ]:
scaler = StandardScaler()
X_train_scaled = scaler.fit_transform(X_train)
X_test_scaled = scaler.fit_transform(X_test)

In [ ]:
model = LogisticRegression()
model.fit(X_train_scaled, Y_train)

LogisticRegression()

In [ ]:
Y_pred = model.predict(X_test_scaled)

In [ ]:
accuracy = accuracy_score(Y_test, Y_pred)
conf_matrix = confusion_matrix(Y_test, Y_pred)
class_report = classification_report(Y_test, Y_pred)

print(f"Accuracy: {accuracy}")
print("Confusion Matrix: ")
print(conf_matrix)
print("Classification Report: ")
print(class_report)

Accuracy: 0.7719277673545967
Confusion Matrix: 
[[ 127 1912]
 [  33 6456]]
Classification Report: 
              precision    recall  f1-score   support

           0       0.79      0.06      0.12      2039
           1       0.77      0.99      0.87      6489

    accuracy                           0.77      8528
   macro avg       0.78      0.53      0.49      8528
weighted avg       0.78      0.77      0.69      8528




The model guessed about 77% of the time. It's better at predicting when people will take term deposits, but not so good at predicting when they won't. This means it sometimes makes mistakes, saying people won't take term deposits when they actually do. The bank might need to find better ways to pick out potential customers for term deposits.